core/src/main/java/org/javarosa/core/model/QuestionDef.java

```diff
@@ -236,6 +231,7 @@ public IFormElement getChild(int i) {
      * @see org.javarosa.core.util.Externalizable#readExternal(java.io.DataInputStream)
      */
     public void readExternal(DataInputStream dis, PrototypeFactory pf) throws IOException, DeserializationException {
+        super.readExternal(dis, pf);
         setID(ExtUtil.readInt(dis));
         binding = (XPathReference)ExtUtil.read(dis, new ExtWrapNullable(new ExtWrapTagged()), pf);
         setAppearanceAttr((String) ExtUtil.read(dis, new ExtWrapNullable(String.class), pf));
```

Comment:
Serialization of `QuestionDef` isn't covered in the javarosa or commcare test suites. Can you look into how hard it would be to add test coverage of `QuestionDef`? This is a pretty straight forward change, but it would still be nice to cover it in the test suite, since serialization-related crashes are sometimes only caught in QA/after release.